streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -67,21 +67,19 @@ public void init(final ProcessorContext context, final StateStore root) {
         streamThread = Thread.currentThread();
     }
 
-    @SuppressWarnings("unchecked")
+    //@SuppressWarnings("unchecked")
     private void initInternal(final ProcessorContext context) {
         this.context = (InternalProcessorContext) context;
-        this.serdes = new StateSerdes<>(ProcessorStateManager.storeChangelogTopic(context.applicationId(), underlying.name()),
-                                        keySerde == null ? (Serde<K>) context.keySerde() : keySerde,
-                                        valueSerde == null ? (Serde<V>) context.valueSerde() : valueSerde);
+        serdes = new StateSerdes<>(
```

Comment:
more of a meta-comment, but do we want to piggyback the work of not using the cache at all if the user specifies `cache max bytes = 0`? I'm guessing not, probably better in a separate PR, but I just wanted to note that.